graylog2-server/src/main/java/org/graylog2/Configuration.java

```diff
@@ -276,6 +276,14 @@ public class Configuration extends CaConfiguration implements CommonNodeConfigur
     @Parameter(value = INSTALL_OUTPUT_BUFFER_DRAINING_MAX_RETRIES, validators = PositiveIntegerValidator.class)
     private int installOutputBufferDrainingMaxRetries = DEFAULT_INSTALL_RETRIES;
 
+
+    @Parameter(value = "timestamp_grace_period", validators = PositiveDurationValidator.class)
```

Comment:
I would greatly prefer this to be a cluster config value that can be edited at runtime instead of requiring a server restart.

In addition to the restarts, if this is configured per server, it can also become inconsistent across a cluster, which is really difficult to debug.

Reply:
30 days also seems very long to me. I think a grace period would exist to cover jitter/issues with clock synchronization of the Graylog nodes, so I would have expected a couple of minutes/some hours, definitely not more than a day. 30 days is in the margin of an unsynchronized clock, where any grace period becomes completely arbitrary and potentially detrimental.

Reply:
Scratch my previous comment. It's not an issue because we talk about future times.

Reply:
Switched to a cached cluster config.
Default value is 2 days to not confuse issues with TZ offsets.